components/PlantCard.js

```diff
@@ -1,15 +1,17 @@
 import styled from "styled-components";
+import PlantFavIcon from "@/public/Icons/plant-line.svg";
+import PlantFavIconFill from "@/public/Icons/plant-fill.svg";
 
-export default function PlantCard({ plant }) {
+export default function PlantCard({ plant, toggleFavourite }) {
   return (
     <Container>
+      <button onClick={() => toggleFavourite(plant.id)}>
```

Comment:
Consider adding an aria-label for the button to make it accessible. This will help screen readers understand its purpose, especially if the icon changes based on state.

```suggestion
      <button aria-label={plant.isFavourite ? "Remove from favorites" : "Add to favorites"} onClick={() => toggleFavourite(plant.id)} >
```